### ISSUE TITLE
[BUG] Repeat Combinator does not work on v0.2.1

### ISSUE THREAD

**Issue by arijit-dasgupta** [genjax-community/genjax#1060]
**Describe the bug**
The `repeat` combinator does not work even after applying [Mirko's fix](https://github.com/probcomp/genjax/pull/1034
)
**To Reproduce**
```python
from genjax import static_gen_fn
import genjax
import jax
key = jax.random.PRNGKey(234)

@static_gen_fn
def func(input):
    x, t = input
    y = genjax.normal(x, 1.0) @ 'y'
    z = genjax.normal(y, 5.0) @ 'z'
    return (z,t+1)

repeat_func = genjax.repeat_combinator(repeats=7)(func)

@static_gen_fn
def outer_func():
    outputs, times = repeat_func((3.2,0)) @ "out"
    return outputs

tr = outer_func.simulate(key,())
```

I end up getting `TypeError: Can't instantiate abstract class RepeatTrace with abstract method get_gen_fn`
The same bug comes up even if I use repeat as a decorator. 

**Execution environment:**
 - OS: Debian
 - Python version: 3.11.8
 - GenJAX version: 0.2.1

**Comment by sritchie** [comment]
Thanks for this, @arijitnoobstar , we will take a look at this this week!

**Comment by femtomc** [comment]
Closed by #1072.
